src/core_plugins/metrics/public/components/custom_color_picker.js

```diff
@@ -0,0 +1,199 @@
+import ChromeFields from 'react-color/lib/components/chrome/ChromeFields';
+import ChromePointer from 'react-color/lib/components/chrome/ChromePointer';
+import ChromePointerCircle from 'react-color/lib/components/chrome/ChromePointerCircle';
+import CompactColor from 'react-color/lib/components/compact/CompactColor';
+import color from 'react-color/lib/helpers/color';
+import shallowCompare from 'react-addons-shallow-compare';
+
+export class CustomColorPicker extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(data) {
+    this.props.onChange(data);
+  }
+
+  render() {
+    const rgb = this.props.rgb;
+    const styles = reactCSS({
+      'default': {
+        picker: {
+          background: '#fff',
+          borderRadius: '2px',
+          boxShadow: '0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)',
+          boxSizing: 'initial',
+          width: '275px',
+          fontFamily: 'Menlo',
+        },
+        saturation: {
+          width: '100%',
+          paddingBottom: '55%',
+          position: 'relative',
+          borderRadius: '2px 2px 0 0',
+          overflow: 'hidden',
+        },
+        Saturation: {
+          radius: '2px 2px 0 0',
+        },
+        body: {
+          padding: '16px 16px 12px',
+        },
+        controls: {
+          display: 'flex',
+        },
+        color: {
+          width: '32px',
+        },
+        swatch: {
+          marginTop: '6px',
+          width: '16px',
+          height: '16px',
+          borderRadius: '8px',
+          position: 'relative',
+          overflow: 'hidden',
+        },
+        active: {
+          absolute: '0px 0px 0px 0px',
+          borderRadius: '8px',
+          boxShadow: 'inset 0 0 0 1px rgba(0,0,0,.1)',
+          background: `rgba(${ rgb.r }, ${ rgb.g }, ${ rgb.b }, ${ rgb.a })`,
+          zIndex: '2',
+        },
+        toggles: {
+          flex: '1',
+        },
+        hue: {
+          height: '10px',
+          position: 'relative',
+          marginBottom: '8px',
+        },
+        Hue: {
+          radius: '2px',
+        },
+        alpha: {
+          height: '10px',
+          position: 'relative',
+        },
+        Alpha: {
+          radius: '2px',
+        },
+        swatches: {
+          display: 'flex',
+          flexWrap: 'wrap',
+          justifyContent: 'center',
+          marginTop: '10px'
+        }
+      },
+      'disableAlpha': {
+        color: {
+          width: '22px',
+        },
+        alpha: {
+          display: 'none',
+        },
+        hue: {
+          marginBottom: '0px',
+        },
+        swatch: {
+          width: '10px',
+          height: '10px',
+          marginTop: '0px',
+        },
+      },
+    }, this.props);
```

Comment:
I might be misunderstanding this, but why not just use regular classes here? This just seems more difficult to read to me. I'd just do:

```
const classes = classnames('custom-color-picker', {
  'custom-color-picker--no-alpha': this.props.disableAlpha
})
```

or something like that (using [`classnames`](https://www.npmjs.com/package/classnames) here, which was extracted from React core).

(I don't have anything against css in JS, though — this is just for consistency with the rest of the code base)